source/application/SGM/SGMAppController.cpp

```diff
@@ -83,6 +84,32 @@ void SGMAppController::shutdown() {
 	AMAppController::shutdown();
 }
 
+void SGMAppController::initializeAmptekView()
+{
+	if(!amptekDetectorView_) {
+
+		amptekDetectorView_ = new CLSAmptekDetailedDetectorView(SGMBeamline::sgm()->amptekSDD1());
+		amptekDetectorView_->buildDetectorView();
+		amptekDetectorView_->setEnergyRange(270,2000);
+		amptekDetectorView_->collapsePeriodTableViews();
+		mw_->addPane(amptekDetectorView_, "Amptek Detector", "Amptek Detector", ":/system-software-update.png");
+		connect(amptekDetectorView_, SIGNAL(resized()), this, SLOT(onAmptekDetectorViewResized()));
+	}
+}
+
+void SGMAppController::onAmptekDetectorViewResized()
+{
+	// Hack solution copied over from the old SGM App Controller.
+	// If this timer is not here it doesn't always resize properly.
+	QTimer::singleShot(100, this, SLOT(resizeToMinimum()));
```

Comment:
Dunno, just copied it over. I could give it a try and see how it behaves without the timer. Now seems like as good a chance as any.